pokemon-example/eva.py

```diff
@@ -375,106 +375,120 @@ async def initialize_state(self, morph_instance: 'MorphInstance') -> Instance[S,
         pass
 
 
-async def run(task: VerifiedTask[S, A, R, T], agent: Agent[S, A, R, T], max_steps: int = 100, 
-        verify_every_step: bool = False, ttl_seconds: Optional[int] = None) -> Tuple[VerificationResult[R], Trajectory[S, A, R, T]]:
+async def initialize_run(task: VerifiedTask[S, A, R, T], agent: Agent[S, A, R, T], ttl_seconds: Optional[int] = None) -> Tuple[Instance[S, T], Trajectory[S, A, R, T], 'MorphInstance']:
     """
-    Run an agent on a task until the task is complete or max_steps is reached.
+       Initialize a run by creating the Morph instance and initial state.
+    
+    Returns:
+        Tuple of (initial_state, trajectory, morph_instance)
```

Comment:
specificaly break up run into initialize and continue steps

and dont eagerly stop instances